### PR TITLE
Support email plugin status sink

### PR DESCRIPTION
**Pull request by bernardjkim**
This PR enables the Email plugin to emit its plugin status.
This change is part of the work required to support Teleport-hosted Email plugins.

### CODE REVIEW

File: integrations/access/email/mailers.go [hunk 1]
```diff
@@ -123,6 +167,38 @@ func (m *SMTPMailer) base36(input uint64) string {
...
+}
+
+// CheckHealth checks the health of the Mailgun service.
+func (m *MailgunMailer) CheckHealth(ctx context.Context) error {
+	ctx, cancel := context.WithTimeout(ctx, mailgunHTTPTimeout)
+	defer cancel()
+
+	msg := m.mailgun.NewMessage(m.sender, "Health Check", "Testing Mailgun API connection...", m.fallbackRecipients...)
+	msg.SetRequireTLS(true)
+	msg.EnableTestMode() // Test message submission without delivering to recpients.
```
Comment (marcoandredinis) [inline integrations/access/email/mailers.go:197]:
```suggestion
	msg.EnableTestMode() // Test message submission without delivering to recipients.
```

File: integrations/access/email/mailers.go [hunk 2]
```diff
@@ -29,42 +30,85 @@ import (
...
 }
 
 // NewSMTPMailer inits new SMTP mailer
-func NewSMTPMailer(c SMTPConfig, sender, clusterName string) Mailer {
+func NewSMTPMailer(c SMTPConfig, sink common.StatusSink, sender, clusterName string) Mailer {
 	dialer := mail.NewDialer(c.Host, c.Port, c.Username, c.Password)
 	dialer.StartTLSPolicy = c.MailStartTLSPolicy
 
-	return &SMTPMailer{dialer, sender, clusterName}
+	return &SMTPMailer{dialer, sender, clusterName, sink}
```
Comment (marcoandredinis) [inline integrations/access/email/mailers.go:76]: Can we please change this constructor to use named fields?

File: integrations/access/email/mailers.go [hunk 3]
```diff
@@ -29,42 +30,85 @@ import (
...
 // NewMailgunMailer inits new Mailgun mailer
-func NewMailgunMailer(c MailgunConfig, sender, clusterName string) Mailer {
+func NewMailgunMailer(c MailgunConfig, sink common.StatusSink, sender, clusterName string, fallbackRecipients []string) Mailer {
 	m := mailgun.NewMailgun(c.Domain, c.PrivateKey)
 	if c.APIBase != "" {
 		m.SetAPIBase(c.APIBase)
 	}
-	return &MailgunMailer{m, sender, clusterName}
+	client := &http.Client{
+		Transport: &statusSinkTransport{
```
Comment (marcoandredinis) [inline integrations/access/email/mailers.go:86]: IIRC, we've recently changed the plugins' http clients to account for environment proxy variables. Ofc, this only makes sense for self-hosted deployments.

File: integrations/access/email/mailers.go [hunk 4]
```diff
@@ -123,6 +167,38 @@ func (m *SMTPMailer) base36(input uint64) string {
...
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, statusEmitTimeout)
+	defer cancel()
+
+	log := logger.Get(ctx)
+	code := http.StatusOK
+	if err != nil {
+		// Returned error is undocumented. Using geneirc error code for all errors.
```
Comment (marcoandredinis) [inline integrations/access/email/mailers.go:182]:
```suggestion
		// Returned error is undocumented. Using generic error code for all errors.
```

**Review by marcoandredinis — APPROVED** [review]

File: integrations/access/email/mailers.go (hunk 3 above)
Comment (bernardjkim) [reply integrations/access/email/mailers.go:86]: Thanks for pointing this out. I wasn't aware of the change. But it actually looks like the `DefaultTransport` already uses `ProxyFromEnvironment` . https://pkg.go.dev/net/http#DefaultTransport

**Review by hugoShaka — APPROVED** [review]

File: integrations/access/email/mailers.go [hunk 5]
```diff
@@ -147,3 +231,29 @@ func (m *MailgunMailer) Send(ctx context.Context, id, recipient, body, reference
...
+	log := logger.Get(req.Context())
+	resp, err := t.RoundTripper.RoundTrip(req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if t.sink != nil {
+		ctx, cancel := context.WithTimeout(req.Context(), statusEmitTimeout)
+		defer cancel()
+
+		status := common.StatusFromStatusCode(resp.StatusCode)
```
Comment (tigrato) [inline integrations/access/email/mailers.go:253]: @bernardjkim 

sorry for the delay. A few months ago, we began adding dedicated status entries for plugins, where we include detailed error information like the error code, a human-readable message, and the original error itself. We’ve implemented this approach for Entra, GitLab, and others.

I suggest doing the same here. A simple "running/not running" status doesn’t provide meaningful insight, especially when running plugins in teleport cloud tenants where users don't have direct access to logs.

Examples of good information:
- can't dial to mailgun
- credentials invalid (expired)
- message rejected...

check 

https://github.com/gravitational/teleport/blob/64922dffc15846a395f765fc524988227e6a3bc2/api/types/types.pb.go#L17107-L17130

Comment (bernardjkim) [reply integrations/access/email/mailers.go:253]: Hey, that is a good point. The generic "running / not running" status isn't all that helpful. I'll work on a follow up PR soon to improve the status entries.

Comment (bernardjkim) [reply integrations/access/email/mailers.go:253]: Tracking in https://github.com/gravitational/teleport/issues/48604 so I don't forget

**Pull request merged**